pytorch_lightning/trainer/training_loop.py

```diff
@@ -91,12 +91,8 @@ def num_optimizers(self):
         return num_optimizers
 
     def should_skip_training(self):
-        if self.trainer.current_epoch >= self.trainer.max_epochs:
+        if self.trainer.current_epoch >= self.trainer.max_epochs or self.trainer.num_training_batches == 0:
             return True
```

Comment:
```suggestion
        return self.trainer.current_epoch >= self.trainer.max_epochs or self.trainer.num_training_batches == 0
```

Reply:
I think this is enough

Reply:
also can you check it `max_steps=0`, does it run?

Reply:
Yup, it does run with `max_steps=0`. Will follow up with a fix for that

Reply:
going to do this in a followup PR?

Reply:
I will follow up after this [PR](https://github.com/PyTorchLightning/pytorch-lightning/pull/5726)